FailedBonusActivationActivity/handler.ts

```diff
@@ -0,0 +1,187 @@
+  "FailedBonusActivationFailure"
+);
+export type FailedBonusActivationFailure = t.TypeOf<
+  typeof FailedBonusActivationFailure
+>;
+
+export const FailedBonusActivationResult = t.taggedUnion("kind", [
+  FailedBonusActivationSuccess,
+  FailedBonusActivationFailure
+]);
+export type FailedBonusActivationResult = t.TypeOf<
+  typeof FailedBonusActivationResult
+>;
+
+type IFailedBonusActivationHandler = (
+  context: Context,
+  input: unknown
+) => Promise<FailedBonusActivationResult>;
+
+/**
+ * Converts a Promise<Either> into a TaskEither
+ * This is needed because our models return unconvenient type. Both left and rejection cases are handled as a TaskEither left
+ * @param lazyPromise a lazy promise to convert
+ *
+ * @returns either the query result or a query failure
+ */
+const fromQueryEither = <R>(
+  lazyPromise: () => Promise<Either<QueryError | Error, R>>
+): TaskEither<Error, R> =>
+  tryCatch(lazyPromise, toError).chain(errorOrResult =>
+    fromEither(errorOrResult).mapLeft(toError)
+  );
+
+/**
+ * Release the lock that was eventually acquired for this request. A release attempt on a lock that doesn't exist is considered successful.
+ *
+ * @param bonusLeaseModel an instance of BonusLeaseModel
+ * @param familyMembers the family of the requesting user
+ *
+ * @returns either a conflict error or the unique hash id of the family
+ */
+const relaseLockForUserFamily = (
+  bonusLeaseModel: BonusLeaseModel,
+  familyUID: FamilyUID
+): TaskEither<UnhandledFailure, FamilyUID> => {
+  return fromQueryEither(() => bonusLeaseModel.deleteOneById(familyUID)).bimap(
+    err =>
+      UnhandledFailure.encode({
+        kind: "UNHANDLED_FAILURE",
+        reason: `Error releasing lock: ${err.message}`
+      }),
+    _ => familyUID
+  );
+};
+
+const updateBonusAsFailed = (
+  bonusActivationModel: BonusActivationModel,
+  bonusActivation: BonusActivationWithFamilyUID
+): TaskEither<UnhandledFailure, RetrievedBonusActivation> => {
+  return fromQueryEither(() => {
+    const bonusToUpdate: NewBonusActivation = {
+      ...bonusActivation,
+      id: bonusActivation.id as BonusCode & NonEmptyString,
+      kind: "INewBonusActivation",
+      status: BonusActivationStatusEnum.FAILED
+    };
+    return bonusActivationModel.createOrUpdate(
+      bonusToUpdate,
+      bonusActivation.id
+    );
+  }).mapLeft(err =>
+    UnhandledFailure.encode({
+      kind: "UNHANDLED_FAILURE",
+      reason: err.message
+    })
+  );
+};
+
+const deleteEligibilityCheck = (
+  eligibilityCheckModel: EligibilityCheckModel,
+  bonusActivation: BonusActivationWithFamilyUID
+): TaskEither<TransientFailure, BonusActivationWithFamilyUID> => {
+  return fromQueryEither(() =>
+    eligibilityCheckModel.deleteOneById(
+      bonusActivation.id as BonusCode & NonEmptyString
+    )
+  )
+    .mapLeft(_ =>
+      // for now I consider this failures as transient, I'll ask a retry to the orchestrator
+      TransientFailure.encode({ kind: "TRANSIENT" })
```

Comment:
just pass through here and log the errore, do not trigger a retry

Reply:
should I ignore this error and fold to the the right path?

Reply:
yep